src/libraries/Common/src/Interop/OSX/Swift.Runtime/UnsafeBufferPointer.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Swift.Runtime
+{
+    // <summary>
+    // Represents Swift UnsafeBufferPointer in C#.
+    // </summary>
+    internal readonly unsafe struct UnsafeBufferPointer<T> where T : unmanaged
```

Comment:
Is this and `UnsafeMutableBufferPointer` APIs we should provide or perhaps manufacture in a source generator during marshal code gen? Perhaps create a default marshaller for `Span<T>`/`ReadOnlySpan<T>`?

Reply:
I can write marshallers for these types, but since the next phase of the Swift interop project is going to move us to directly calling the Swift APIs anyway, I don't think we'd really save much by adding in the marshallers now.

Reply:
> the next phase of the Swift interop project is going to move us to directly calling the Swift APIs anyway

Sounds good. Thanks.